### ISSUE TITLE
Remove deprecated key 'root' from node's env

### ISSUE THREAD

**Issue by eight04** [eslint/eslint#9144]
**Tell us about your environment**

* **ESLint Version:** 4.5.0
* **Node Version:** 7.9.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```cmd
eslint --no-eslintrc --env es6,node --rule "no-undef: 2" test.js
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
class Test {
	constructor(root) {
		this.root = root;
	}
	doSomething() {
		console.log(`${root}`);
	}
}

```

**What did you expect to happen?**
```
  6:18  error  'root' is not defined     no-undef
```

**What actually happened? Please include the actual, raw output from ESLint.**
Nothing is reported, because `root` is a global variable in node env.

However, global `root` is a [deprecated feature](https://nodejs.org/api/deprecations.html#deprecations_dep0016_global_root). By removing `root` from node's env, it should help those developers that accidentally access global `root` variable.

**Comment by kaicataldo** [comment]
Unfortunately, ESLint can't make any assumptions about which version of Node someone is using. This is the package we use: https://github.com/sindresorhus/globals/blob/master/globals.json#L867

There are third party plugins such as https://www.npmjs.com/package/eslint-plugin-node that will actually do feature-checking based on the Node version, and something like that could be a good place for a feature like this.

Alternatively, are you suggesting that ESLint allows specification of Node version? That's an interesting thought, though it's not a trivial thing to implement.

**Comment by eight04** [comment]
It seems that this issue should be moved to https://github.com/sindresorhus/globals.

I understand those platform-specified rules should be handled by plugins. However, I don't know if env global is in this case. Are those globals maintained by ESLint team? For example, Greasemonkey is working on version 4.0 which might add/remove some globals too (https://github.com/greasemonkey/greasemonkey/issues/2484).

BTW, https://github.com/mysticatea/eslint-plugin-node looks good. I'll definitely try it.

**Comment by eight04** [comment]
NVM, I saw that those are located in globals.json.
https://github.com/sindresorhus/globals/blob/890bcfe6e052739ce2d4dea038589f9243cf9b17/globals.json#L1277

**Comment by platinumazure** [comment]
@eight04 By the way, you could use `no-restricted-globals` to enforce that `root` is not used as a global in your code.
